examples/algorithms/plot_metric_closure.py

```diff
@@ -0,0 +1,76 @@
+"""
+==============
+Metric Closure
+==============
+
+The *metric closure* of a graph is the complete graph in which
+each edge weight corresponds to the shortest-path distance
+between two nodes in the original graph.
+
+This example demonstrates the *unweighted* version of the metric closure,
+constructed using :func:`networkx.all_pairs_shortest_path_length`.
+Here, each edge is assumed to have weight 1, so distances correspond to
+the minimum number of hops between nodes.
+
+For weighted graphs, the metric closure can be obtained instead with
+:func:`networkx.all_pairs_dijkstra_path_length`, which computes distances
+using the minimum total edge weights along paths.
+"""
+
+import networkx as nx
+import matplotlib.pyplot as plt
+
+# Creating five nodes from index 0 to 4
+G = nx.Graph()
+G.add_nodes_from(range(5))
+G.add_edge(0, 1)
+G.add_edge(0, 4)
+G.add_edge(1, 3)
+G.add_edge(1, 5)
+G.add_edge(2, 3)
+G.add_edge(2, 4)
+G.add_edge(2, 5)
+G.add_edge(3, 4)
+G.add_edge(3, 5)
+G.add_edge(4, 5)
```

Comment:
For the sake of reducing cognitive load, perhaps a built-in example could work? E.g. `nx.house_graph`, which gives a very similar result.